Jenkinsfile.nightly

```diff
@@ -36,6 +36,7 @@ pipeline {
         GOPATH = "${WORKSPACE}"
         TESTDIR = "${WORKSPACE}/${PROJ_PATH}/test"
         SERVER_BOX = "cilium/ubuntu"
+        NIGHTLY_ID =  "${BUILD_NUMBER}"
```

Comment:
Is BUILD_NUMBER the the jenkins build number? I wonder if that can repeat if we delete builds etc.
Could we make this the date? (e.g. 20180705) so it's easier to refer to the images?

Reply:
AFAIK jenkins build number cannot be repeated even if we delete builds, but maybe @eloycoto can confirm?

Reply:
Oh, I meant the inevitable time jenkins crashes and we lose whatever database it keeps. In any case, the build numbers are rather opaque compared to some datestamp for nightlies. They (nightlies) have a natural name because they are fixed in time.

Reply:
@raybejjani I decided to use both, as it is possible that we will get two nightly builds on the same date and end up overwriting previous tag.

Reply:
> it's not very straightforward without additional plugins

I was worried you'd say that. I guess we can't shell out or use an environment variable, right?

> it is possible that we will get two nightly builds on the same date and end up overwriting previous tag

Good point. I guess the later one will be the "latest" for that day then? This seems reasonable. +1

Reply:
Build Id is ok, if we change the job names we can change the start number. ;-)

Reply:
Sure, but the nightly images are intended to be used. A random number is less useful when talking about them than the actual date (the point about multiple builds holds, however).